sdk/core/core-paging/src/getPagedAsyncIterator.ts

```diff
@@ -28,26 +30,38 @@ export function getPagedAsyncIterator<
     },
     byPage:
       pagedResult?.byPage ??
-      ((settings?: PageSettings) => {
+      (((settings?: PageSettings) => {
         const { continuationToken, maxPageSize } = settings ?? {};
-        return getPageAsyncIterator(pagedResult as PagedResult<TPage, PageSettings, TLink>, {
-          pageLink: continuationToken as unknown as TLink | undefined,
-          maxPageSize,
-        });
-      }),
+        return getPageAsyncIterator(
+          pagedResult as PagedResult<TPage, PageSettings, TLink, TElement>,
+          {
+            pageLink: continuationToken as unknown as TLink | undefined,
+            maxPageSize,
+          }
+        );
+      }) as unknown as (settings?: TPageSettings) => AsyncIterableIterator<TPage>),
```

Comment:
also curious if we can avoid this explicit cast

Reply:
This change is unrelated to the PR but without it you will get annoying squiggly red lines in vscode because of added smarts in the recent TS version. This code instantiates the `byPage` function with a default implementation if none was provided. However, the `settings` parameter is allowed to be of a generic type but the default implementation uses `PageSettings`. If the customer instantiates the generic to anything else, the typing will not be consistent.

However, in the current design's defense, the generic defaults to `PageSettings` and it doesn't make sense for the customer to instantiate it without providing an implementation.

Reply:
Could we enforce the relationship between the two somehow with extends on the generic parameter?

Reply:
IIRC, we didn't do that mainly so we can support the operations that use composite continuation tokens (an example in documents-search came up but I don't remember it) or non-string continuation tokens (`continuationToken` is of type string in `PageSettings`). However, I admit the latter case could be solved by re-defining `continuationToken` with the desired type.